app/backend/src/couchers/servicers/communities.py

```diff
@@ -144,6 +145,86 @@ def ListCommunities(self, request, context, session):
             next_page_token=encrypt_page_token(str(offset + page_size)) if len(nodes) > page_size else None,
         )
 
+    def SearchCommunities(self, request, context, session):
+        raw_q = (request.query or "").strip()
+        if not raw_q:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "query_is_empty")
+        if len(raw_q) < 3:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, "query_too_short")
+
+        req_size = request.page_size or MAX_PAGINATION_LENGTH
+        page_size = max(1, min(MAX_PAGINATION_LENGTH, req_size))
```

Comment:
See comment on page_size being a `uint`, then there's no need to do the maxing here.